files/default/neutron-ha-tool.py

```diff
@@ -1058,6 +1025,95 @@ def pick(self):
         )
 
 
+class RemoteRouterNsCleanup(object):
+
+    def __init__(self, host, routerid):
+        self.target_host = host
+        self.namespace = "qrouter-" + routerid
+        self.ssh_client = paramiko.SSHClient()
+        self.ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.ssh_client.load_system_host_keys()
+        self.ssh_client.connect(self.target_host, timeout=10)
+        self.timeout = 10
+        self.netns_del = "ip netns delete " + self.namespace
+        self.netns_pids = "ip netns pids " + self.namespace
+        self.netns_list = "ip netns list"
+
+    def _simple_ssh_command(self, command):
+        # Note, that when get_pty is True, paramiko will never return anything
+        # on the stderr channel, that's why we ignore it here. (stderr output
+        # will endup in the stdout channel)
+        _, stdout, _ = self.ssh_client.exec_command(command,
+                                                    timeout=10,
+                                                    get_pty=True)
+        out_lines = stdout.readlines()
+        rc = stdout.channel.recv_exit_status()
+        return [rc, out_lines]
+
+    def _namespace_exists(self):
+        rc, out_lines = self._simple_ssh_command(self.netns_list)
+        return self.namespace in [ns.strip() for ns in out_lines]
+
+    def _get_namespace_pids(self):
+        rc, out_lines = self._simple_ssh_command(self.netns_pids)
+        if rc:
+            if out_lines and "No such file or directory" in out_lines[0]:
+                # Assume the namespace was delete meanwhile
+                return [0, []]
```

Comment:
I am not sure that this is OK, I think that you want to return an empty array, like in the success case.